src/main/java/com/dev/cinema/security/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.dev.cinema.security.impl;
+
+import com.dev.cinema.exceptions.AuthenticationException;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.User;
+import com.dev.cinema.security.AuthenticationService;
+import com.dev.cinema.service.UserService;
+import com.dev.cinema.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User userFromDB = userService.findByEmail(email).orElseThrow(() ->
+                new AuthenticationException("User with such email not found"));
```

Comment:
let's make userService return user or null, not Optional

Reply:
done